src/python/m5/SimObject.py

```diff
@@ -1236,6 +1248,14 @@ def getCCParams(self):
                 else:
                     for v in value:
                         getattr(cc_params, param).append(v)
+            elif isinstance(self._params[param], DictParamDesc):
+                assert isinstance(value, dict)
+                dic = getattr(cc_params, param)
+                assert not len(dic)
+                if isinstance(dic, dict):
+                    setattr(cc_params, param, dict(value))
+                else:
+                    raise TypeError("Shouldn't happen")
```

Comment:
Can you explain to the user what shouldn't happen? Maybe `TypeError("Must provide dictionary for dictionary param {name()}")` or something like that.